### PR TITLE
interfaces: posix-mq: allow specifying message queue paths as an array

### PR DESCRIPTION
**Pull request by IsaacJT**
Posix message queue paths can now either be specified as a single string or an array of strings. This allows consolidating many slots with different paths and identical permissions into a single slot.

E.g.:

```
slots:
  test-string:
    interface: posix-mq
    path: /test-default

  test-path-array:
    interface: posix-mq
    path:
      - /test-array-1
      - /test-array-2
      - /test-array-3
```

### CODE REVIEW

**Review by mardy — COMMENTED** [review]
Hi Isaac, thanks for this! The new code looks good, but I've noticed something that seems to be wrong in the existing code, so this could be a good opportunity to fix it.

File: interfaces/builtin/posix_mq.go
Comment (IsaacJT) [reply interfaces/builtin/posix_mq.go:199]: Good point! It first checks it against the regex afterwards. I'll add in a zero-length check

**Review by mardy — APPROVED** [review]
LGTM, thanks for the update! (please don't merge it yet, we need Samuele's review)

File: interfaces/builtin/posix_mq.go
Comment (pedronis) [inline interfaces/builtin/posix_mq.go:290]: this comment should tell if this is for a slot or plug. Also because as written would think than name is the queue name but it's actually the slot/plug name.

**Review by pedronis — COMMENTED** [review]
small comment

File: interfaces/builtin/posix_mq.go
Comment (mvo5) [reply interfaces/builtin/posix_mq.go]: I added this now in [8cf768e](https://github.com/snapcore/snapd/pull/11928/commits/8cf768e16374bd69f2bc0677b26d4f84d30357b9) in a slightly naive way, let me know if you want me to tweak this (happy to do so).

**Comment by pedronis** [comment]
the last commit looks reasonable to me. I can't do a full approval review though, this needs a couple of 2nd reviews from other people

**Review by MiguelPires — APPROVED** [review]
LGTM, thanks

File: interfaces/builtin/posix_mq_test.go
Comment (mvo5) [reply interfaces/builtin/posix_mq_test.go:74]: I added this now.

**Review by mvo5 — APPROVED** [review]
Thank you! This is fine, one suggestion for a followup inline

File: interfaces/builtin/posix_mq.go
Comment (mvo5) [reply interfaces/builtin/posix_mq.go:182]: I added the followup under https://github.com/snapcore/snapd/pull/11992

**Pull request merged**
